pkg/alertmanager/amcfg.go

```diff
@@ -72,9 +76,16 @@ func newConfigGenerator(logger log.Logger, store *assets.Store) *configGenerator
 
 func (cg *configGenerator) generateConfig(
 	ctx context.Context,
+	a *v1.Alertmanager,
 	baseConfig alertmanagerConfig,
 	amConfigs map[string]*monitoringv1alpha1.AlertmanagerConfig,
 ) ([]byte, error) {
+	versionStr := a.Spec.Version
+	if versionStr == "" {
+		versionStr = operator.DefaultAlertmanagerVersion
+	}
+
+	version, _ := semver.ParseTolerant(versionStr)
```

Comment:
Since the `generateConfig` function already returns an error in its signature, can we handle it here and return it when version parsing fails?

Reply:
Sure thing - will update